### PR TITLE
feat: change the file structure

### PR DESCRIPTION
**Pull request by DiogoRDuarte**
## Because
- Some files add names whose purpose was not obvious. Their placement in the file structure was, at times, not the most suitable one too. Both points apply to functions.

### Refactor
- Moves the CSS inside the "assets" folder
- Changed the names of certain files to make them for descriptive (e.g. DescriptionChanger.ts)
- Moved functions into files that seem more suitable (e.g. xSetter -> NavigationKeyHandler.ts)

### Tests
- Fixes semantic issues with divs inside table rows

### CODE REVIEW

File: src/AutoVizuA11y.tsx [hunk 1]
```diff
@@ -262,40 +284,59 @@ const AutoVizuA11y = ({
...
 
 		//creates the ShortcutGuide
 		createShortcutGuide();
 	}, [ref]);
 
-	//sets the appropriate navigation keys in the ShortcutGuide
+	// sets the appropriate navigation keys in the ShortcutGuide
 	function handleNav(event: React.KeyboardEvent<HTMLDivElement>) {
-		levelNav(event, ref);
+		navigationKeyHandler(
```
Comment (JoaoTMDias) [inline src/AutoVizuA11y.tsx:296]: As a good measure, usually we try to at, at most, 3 params in a function definitions.
If we have more that 3, then we convert those 3 into a single object.

It makes code more readable and easier to maintain.

File: src/AutoVizuA11y.tsx [hunk 2]
```diff
@@ -262,40 +284,59 @@ const AutoVizuA11y = ({
...
 
 	//sets the appropriate navigation keys and shortcuts in the charts and data
-	function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
-		let alertDiv: HTMLElement | null = ref.current?.getElementsByClassName(
-			"a11y_alert",
-		)[0] as HTMLElement;
-		levelNav(event, ref, alertDiv, selectorType, multiSeries, nextSeries, series, selectedSeries);
-		let numberAux = keyHandler(
+	function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>, alertDiv: HTMLDivElement) {
+		let numberAux = navigationKeyHandler(
```
Comment (JoaoTMDias) [inline src/AutoVizuA11y.tsx:313]: Same thing here.

File: src/AutoVizuA11y.tsx [hunk 3]
```diff
@@ -262,40 +284,59 @@ const AutoVizuA11y = ({
...
+			elements,
+			alertDiv,
+			selectedSeries,
+			series,
 			selectorType,
+			multiSeries,
+			nextSeries,
+		);
+		setNumber(numberAux);
+		insightsKeyHandler(
```
Comment (JoaoTMDias) [inline src/AutoVizuA11y.tsx:327]: And here

File: src/AutoVizuA11y.tsx [hunk 4]
```diff
@@ -305,7 +346,10 @@ const AutoVizuA11y = ({
 	return (
 		<div
 			ref={ref}
-			onKeyDown={handleKeyDown}
+			onKeyDown={(event) => {
```
Comment (JoaoTMDias) [inline src/AutoVizuA11y.tsx:349]: Can this `alertDiv` query be made inside the `handleKeyDown` function? If we are accessing a ref that means that it is also available inside that callback, right?

File: src/AutoVizuA11y.tsx [hunk 5]
```diff
@@ -314,7 +358,9 @@ const AutoVizuA11y = ({
 				style={{ textIndent: "-10000px" }}
 				className="a11y_desc visually-hidden"
 				data-testid="a11y_desc"
-				onFocus={handleFocus}
+				onFocus={() => {
```
Comment (JoaoTMDias) [inline src/AutoVizuA11y.tsx:361]: Same thing here. Can't we pass the function reference here and actually get the value of `alertDiv` inside of it?

File: src/ShortcutGuide.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ const ShortcutGuide: React.FC = () => {
 
 	return (
 		<>
-			<div className="a11y_modal" data-testid="a11y_modal" role="button">
+			<div className="a11y_modal" data-testId="a11y_modal" role="button">
```
Comment (JoaoTMDias) [inline src/ShortcutGuide.tsx:25]: The correct attribute value is `data-testid`, why was this changed?

File: src/components/insights/InsightsKeyHandler.ts [hunk 7]
```diff
@@ -31,60 +23,30 @@ type AutoDescriptionsProps = {
  * @export
  * @param {string} type
```
Comment (JoaoTMDias) [inline src/components/insights/InsightsKeyHandler.ts:24]: This is a nitpick, but we do not need jsdoc typing when we use typescript, seems like overkill to me.

File: src/components/insights/OverallComparer.ts [hunk 8]
```diff
@@ -42,7 +42,7 @@ export function overallComparer(
 			}, 1000);
 			return;
 		}
-		alertDiv.textContent = comparer(arrayConverted, focusedData);
+		alertDiv.textContent = messagecreator(arrayConverted, focusedData);
```
Comment (JoaoTMDias) [inline src/components/insights/OverallComparer.ts:45]: Is this correct? Shouldn't it be `messageCreator` (with a capital C)

File: src/components/navigation/JumpX.ts [hunk 9]
```diff
@@ -16,7 +16,7 @@
  * @param {string[]} series
  * @return {void}
  */
-export function jumpXpoints(
+export function jumpXelements(
```
Comment (JoaoTMDias) [inline src/components/navigation/JumpX.ts:19]: Naming here is also not very consistent.
Shouldn't it be `jumpXElements`? (with capital E)
Or use a longer name like `jumpForXElements`, for instance

File: src/components/navigation/NavigationKeyHandler.ts [hunk 10]
```diff
@@ -7,32 +7,51 @@
...
  * @param {(string | undefined)} [multiSeries]
  * @param {() => void} [nextSeries]
- * @param {string[]} [series]
- * @param {string} [selectedSeries]
- * @return {void}
+ * @return {number} Number of points being jumped at a time inside the wrapped chart.
  */
-export function levelNav(
-	event: React.KeyboardEvent<HTMLDivElement>,
+export function navigationKeyHandler(
```
Comment (JoaoTMDias) [inline src/components/navigation/NavigationKeyHandler.ts:31]: Read me comment above on the number of arguments.
The more arguments a function has, the less maintainable it becomes and more prone to errors.

File: src/components/navigation/NavigationKeyHandler.ts [hunk 11]
```diff
@@ -41,24 +60,24 @@ export function levelNav(
 			document.activeElement?.classList.contains("a11y_row") ||
 			document.activeElement?.id === "guide_close"
 		) {
-			return;
+			return number;
 		}
 		if (document.activeElement?.classList.contains("a11y_desc") && alertDiv) {
 			alertDiv.textContent = "You can only change series while focused on a data point";
 			setTimeout(() => {
```
Comment (JoaoTMDias) [inline src/components/navigation/NavigationKeyHandler.ts:67]: Checkout the [`wait` function on the `@feedzai/js-utilities` package.](https://feedzai.github.io/js-utilities/docs/functions/DOM/wait)
It is a promise that wraps around a timeout, but makes your code look cleaner.

**Review by JoaoTMDias — CHANGES_REQUESTED** [review]
Thank for the PR, @DiogoRDuarte!
I've left a couple of comments regarding code structure and documentation.

You can find more info on [clean javascript code here.](https://github.com/ryanmcdermott/clean-code-javascript/tree/master).
Hope it helps, keep up the good work :)

File: src/AutoVizuA11y.tsx (hunk 4 above)
Comment (DiogoRDuarte) [reply src/AutoVizuA11y.tsx:355]: The alertDiv logic will be changed in a subsequent PR, resolving this issue!

File: src/AutoVizuA11y.tsx (hunk 5 above)
Comment (DiogoRDuarte) [reply src/AutoVizuA11y.tsx:367]: Same point as before :D

File: src/components/insights/OverallComparer.ts (hunk 8 above)
Comment (DiogoRDuarte) [reply src/components/insights/OverallComparer.ts:45]: This logic also changes in a following PR :)

File: src/components/navigation/NavigationKeyHandler.ts (hunk 11 above)
Comment (DiogoRDuarte) [reply src/components/navigation/NavigationKeyHandler.ts:68]: I have identified this and other places in the code where it makes sense to use js-utilities functions. I believe it would be best to submit a separate PR for these changes in the future.

**Review by JoaoTMDias — APPROVED** [review]
